h2o-automl/src/test/java/ai/h2o/automl/AutoMLKeepCVPreds.java

```diff
@@ -0,0 +1,55 @@
+package ai.h2o.automl;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.Key;
+import water.fvec.Frame;
+
+import static org.junit.Assert.assertTrue;
+
+public class AutoMLKeepCVPreds extends water.TestUtil {
+
+  @BeforeClass
+  public static void setup() {
+    stall_till_cloudsize(1);
+  }
+
+  long autoMLSeed = 2345;
+
+  int numberOfModelsToCompareWith = 1;
+  
+  Algo[] excludeAlgos = {Algo.DeepLearning, Algo.DRF, Algo.GLM /*Algo.XGBoost */, Algo.GBM, Algo.StackedEnsemble};
+//  Algo[] excludeAlgos = {/*Algo.DeepLearning,*/ Algo.DRF, Algo.GLM , Algo.XGBoost , Algo.GBM, Algo.StackedEnsemble};
+
+  @Test
+  public void keep_cv_predictions_parameter_is_being_ignored_test() {
+    AutoML aml = null;
+    Frame fr = null;
+
+    String responseColumnName = "survived";
+    AutoMLBuildSpec autoMLBuildSpec = new AutoMLBuildSpec();
+
+    fr = parse_test_file(Key.make("titanic_test_parsed"), "smalldata/gbm_test/titanic.csv");
+
+    autoMLBuildSpec.input_spec.training_frame = fr._key;
+    autoMLBuildSpec.build_control.nfolds = 5;
+    autoMLBuildSpec.input_spec.response_column = responseColumnName;
+    autoMLBuildSpec.build_models.exclude_algos = excludeAlgos;
+
+    autoMLBuildSpec.build_control.project_name = "1234";
+    autoMLBuildSpec.build_control.stopping_criteria.set_max_models(numberOfModelsToCompareWith);
+    autoMLBuildSpec.build_control.stopping_criteria.set_seed(autoMLSeed);
+    autoMLBuildSpec.build_control.keep_cross_validation_models = false;
+    autoMLBuildSpec.build_control.keep_cross_validation_predictions = false;
+
+    long start1 = System.currentTimeMillis();
+    aml = AutoML.startAutoML(autoMLBuildSpec);
+    aml.get();
+
+    Key[] cross_validation_predictions = aml.leader()._output._cross_validation_predictions;
+    assertTrue(cross_validation_predictions == null);
+        
```

Comment:
@deil87 @ledell 
this passes the test:
```
    assertNotNull(cross_validation_predictions);
    assertEquals(autoMLBuildSpec.build_control.nfolds, cross_validation_predictions.length);
    for (Key k : cross_validation_predictions) {
      assertNull(k.get());
    }
```

At some point, I had the intention to nullify this array (same for `cross_validation_models`) as it wasn't done before anyway, but after a discussion with @michalkurka we decided to leave it as it was: delete the keys from DKV but leave the output as an array with the keys, even if they don't reference any object anymore: cf. https://github.com/h2oai/h2o-3/pull/2724#discussion_r210131092

Reply:
@seb-h2o thanks for the reference to discussion. Do we have inconsistency now? because I believe I have seen that some models return null if parameter is set to false.

Reply:
@deil87 it can probably be null if CV was disabled, but should be consistently an `array of dereferenced keys` otherwise